include/chopper/layout/hibf_statistics.hpp

```diff
@@ -136,6 +136,23 @@ class hibf_statistics
                   << "#Total HIBF size no correction:" << to_formatted_BF_size(total_size_no_corr) << "\n\n";
     }
 
+    //!\brief Return the total corrected size of the HIBF in bytes
+    size_t get_total_hibf_memory_size()
+    {
+        if (summaries.empty())
+            gather_statistics(top_level_ibf, 0);
+        
+        size_t total_size = 0;
+        
+        // go through each level and collect the memory sizes
+        for (auto const & [level, s] : summaries)
+        {
+            total_size += std::reduce(s.ibf_mem_size.begin(), s.ibf_mem_size.end());
+        }
```

Comment:
```suggestion
        for (auto const & [level, summary] : summaries)
        {
            (void) level;
            total_size += std::reduce(summary.ibf_mem_size.begin(), summary.ibf_mem_size.end());
        }
```

Will probably complain about `level` being unused